### PR TITLE
AB#2583: deploy autoscaler via helm

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- Deploy autoscaler via helm. Same pattern as with the last PRs. 
- Minor formatting in ccm/azure-daemonset.yaml.

### Additional info
- [Test run #228](https://github.com/edgelesssys/constellation/actions/runs/3385907366): 1.24, azure, autoscaling payload
- [Test run #229](https://github.com/edgelesssys/constellation/actions/runs/3385915266): 1.24, gcp, autoscaling payload

### CODE REVIEW

File: cli/internal/helm/testdata/Azure/constellation-services/charts/ccm/templates/azure-daemonset.yaml [hunk 1]
```diff
@@ -16,31 +15,32 @@ spec:
...
-        - name: azureconfig
-          mountPath: /etc/azure
-          readOnly: true
+        - name: cloud-controller-manager
+          image: ccmImageForAzure
+          command:
+          - cloud-controller-manager
+          - --cloud-provider=azure
+          - --leader-elect=true
+          - --cluster-cidr=192.0.2.0/24
```
Comment (daniel-weisse) [inline cli/internal/helm/testdata/Azure/constellation-services/charts/ccm/templates/azure-daemonset.yaml:24]: This value didn't exist in the previous chart and is causing the unit tests to fail.
Should it be herer?

**Review by daniel-weisse — COMMENTED** [review]
Code LGTM
There is a mismatch between the expected templates and the templates generated by the code, please check it

File: cli/internal/helm/testdata/Azure/constellation-services/charts/ccm/templates/azure-daemonset.yaml (hunk 1 above)
Comment (derpsteb) [reply cli/internal/helm/testdata/Azure/constellation-services/charts/ccm/templates/azure-daemonset.yaml:24]: Ah I didn't realize someone changed sth, thought it was just me formatting weirdly :D

**Review by malt3 — APPROVED** [review]
Code LGTM!

**Review by malt3 — APPROVED** [review]

**Pull request merged**